repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -366,11 +366,21 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         //        .map(l -> new LimitedReadInputStream(l, request.getInputStream()))
         //        .orElseGet(request.getInputStream());
 
-        final HttpServletRequestWrapper wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request,
-                new BufferedServletInputStream(requestBodyInputStream));
         final HttpServletResponseWrapper wrappedResponse = new HttpServletResponseWrapper((HttpServletResponse) response,
                 ResponseMode.MUTABLE,
                 ResponseMode.MUTABLE);
+        final BufferedServletInputStream bufferedInputStream = new BufferedServletInputStream(requestBodyInputStream);
+        HttpServletRequestWrapper wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request,
+                bufferedInputStream);
+
+        // Since getParameterMap may read the body, we must reset the InputStream so that we aren't stripping
+        // the body when form parameters are sent.
+        bufferedInputStream.mark(Integer.MAX_VALUE);
+        wrappedRequest.setAttribute("http://openrepose.org/queryParams", wrappedRequest.getParameterMap());
```

Comment:
I legitimately don't know. I figured we must have had a reason, but I certainly can't think of what it was...